### DISCUSSION TITLE
How do you manually move a job to the failed state and not let RQ retry?

### DISCUSSION THREAD

**Discussion by yawhide** [rq/rq#1878] [Q&A]
Hello,

I am trying to use the on_failure callback to prevent a job from retrying. Is there a better way than setting the job's `retry_count` to 0?
I cant find any documentation or an method I can call to "prevent a job from retrying".

My use case I am trying to solve is to prevent a job from retrying if it fails due to a JobTimeout exception.

thanks

**Comment by selwin** [comment]
> My use case I am trying to solve is to prevent a job from retrying if it fails due to a JobTimeout exception.

This is a good point, we should probably implement something like `Retry(max=3, unless=[JobTimeout, MyOtherException])`.

**Comment by robhudson** [comment]
I looked into this as well. One option would be to subclass the `Worker` or `SimpleWorker` class and override the `handle_job_failure` method, adding your logic there.

Example:
```python
class MyWorker(Worker):
    def handle_job_failure(self, job: 'Job', queue: 'Queue', started_job_registry=None, exc_string=''):
        if <logic-to-abort-job-retries>:
            self._stopped_job_id = job.id
            job.retries_left = None
        super().handle_job_failure(job, queue, started_job_registry, exc_string)
```

**Comment by selwin** [comment]
Besides implementing customizable exceptions like `Retry(max=3, unless=[JobTimeout, MyOtherException])`, we can also introduce an RQ specific `FailedJobError` (better name suggestions welcome) which when raised will be moved to `FailedJobRegistry` without being retried.

I'm happy to accept a PR for both :)
